In [1]:
%matplotlib inline
import pandas as pd

In [2]:
cast='C:/Users/PC PC/Downloads/cast.csv'
cast = pd.read_csv(cast)
cast.head()

title  year      name   type                character     n
0        Closet Monster  2015  Buffy #1  actor                  Buffy 4   NaN
1       Suuri illusioni  1985    Homo $  actor                   Guests  22.0
2   Battle of the Sexes  2017   $hutter  actor          Bobby Riggs Fan  10.0
3  Secret in Their Eyes  2015   $hutter  actor          2002 Dodger Fan   NaN
4            Steve Jobs  2015   $hutter  actor  1988 Opera House Patron   NaN

### How many movies are listed in the titles dataframe?

In [3]:
titles='C:/Users/PC PC/Desktop/DS Assignment/pythonS9/titles.csv'
titles=pd.read_csv(titles)
titles.head()

title  year
0           The Rising Son  1990
1  The Thousand Plane Raid  1969
2         Crucea de piatra  1993
3                  Country  2000
4               Gaiking II  2011

In [4]:
print("No.Of Movies - ",len(titles))

No.Of Movies -  230621


Total number of movies in our data set are - 230621

### What are the earliest two films listed in the titles dataframe?

In [5]:
max_years=titles.nlargest(2,'year')
print(max_years)

                    title  year
101854  Settling of Score  2026
228030         Gods Heart  2025


### How many movies have the title "Hamlet"?

In [6]:
print(len(titles[titles.title=='Hamlet']))

18


From the Data set 18 movies have the title name "Hamlet"

### How many movies are titled "North by Northwest"?

In [7]:
print(len(titles[titles.title=='North by Northwest']))

1


The is only one movie with the title "North by Northwest"

### When was the first movie titled "Hamlet" made?

In [8]:
print("The first movie titled 'Hamlet' in the year -",titles[titles.title=="Hamlet"].year.min())


The first movie titled 'Hamlet' in the year - 1910


### List all of the "Treasure Island" movies from earliest to most recent.

In [9]:
titles[titles.title=='Treasure Island'].sort_values(by='year')

title  year
204540  Treasure Island  1918
50896   Treasure Island  1920
189691  Treasure Island  1934
96259   Treasure Island  1950
59186   Treasure Island  1972
110576  Treasure Island  1973
203909  Treasure Island  1985
178072  Treasure Island  1999

The list of Treasure Island movies from the year 1918 to 1999

# How many movies were made in the year 1950?

In [10]:
year=titles[titles['year']==1950]
print("No.of Movies Released in 1950-",year['title'].nunique())

No.of Movies Released in 1950- 1093


### How many movies were made in the year 1960?

In [11]:
year=titles[titles['year']==1960]
print("No.of Movies Released in 1950-",year['title'].nunique())

No.of Movies Released in 1950- 1496


In [12]:
print(titles.iloc[1960])

title    Hell Squad
year           1986
Name: 1960, dtype: object


In [13]:
movies=titles.iloc[1950:1959]
print(movies.count())

title    9
year     9
dtype: int64


### In what years has a movie titled "Batman" been released?

In [14]:
titles[titles.title=='Batman']

title  year
55577  Batman  1989
74047  Batman  1943

The movie titled "Batman" released in the year 1989 nad 1943

### How many roles were there in the movie "Inception"?

In [15]:
print("No.Of Roles In the Movie Inception are- ",len(cast[cast.title=='Inception']))

No.Of Roles In the Movie Inception are-  77


### How many roles in the movie "Inception" are NOT ranked by an "n" value?

In [16]:
print("No. of Roles that are not ranked in the movie Iception - ",len(cast[cast.title=='Inception']) & len(cast[cast.n.isnull()]))

No. of Roles that are not ranked in the movie Iception -  8


### But how many roles in the movie "Inception" did receive an "n" value?

In [17]:
print("No. of Roles that are  ranked in the movie Inception - ",len(cast[cast.title=='Inception']) & len(cast[cast.n.notnull()]))

No. of Roles that are  ranked in the movie Inception -  0


In [18]:
len(cast[cast.title=='Inception']) & len(cast[cast.n.notnull()])

0

### Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

In [19]:
cast[(cast.title=='North by Northwest') & (cast.n.notnull())].sort_values(by='n')

title  year                  name     type  \
840374   North by Northwest  1959            Cary Grant    actor   
3361396  North by Northwest  1959       Eva Marie Saint  actress   
1404704  North by Northwest  1959           James Mason    actor   
3024085  North by Northwest  1959   Jessie Royce Landis  actress   
343701   North by Northwest  1959        Leo G. Carroll    actor   
2923056  North by Northwest  1959  Josephine Hutchinson  actress   
1634552  North by Northwest  1959           Philip Ober    actor   
1229083  North by Northwest  1959         Martin Landau    actor   
2357939  North by Northwest  1959         Adam Williams    actor   
1746383  North by Northwest  1959          Edward Platt    actor   
642085   North by Northwest  1959     Robert Ellenstein    actor   
2212083  North by Northwest  1959          Les Tremayne    actor   
447893   North by Northwest  1959       Philip Coolidge    actor   
1454363  North by Northwest  1959         Patrick McVey    actor   
198496   North by Northwest  1959          Edward Binns    actor   
1334726  North by Northwest  1959             Ken Lynch    actor   

                           character     n  
840374            Roger O. Thornhill   1.0  
3361396                  Eve Kendall   2.0  
1404704              Phillip Vandamm   3.0  
3024085              Clara Thornhill   4.0  
343701                 The Professor   5.0  
2923056                Mrs. Townsend   6.0  
1634552              Lester Townsend   7.0  
1229083                      Leonard   8.0  
2357939                     Valerian   9.0  
1746383              Victor Larrabee  10.0  
642085                         Licht  11.0  
2212083                   Auctioneer  12.0  
447893                     Dr. Cross  13.0  
1454363               Sergeant Flamm  14.0  
198496                Captain Junket  15.0  
1334726  Charley - Chicago Policeman  16.0

### Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

In [20]:
cast[(cast.title=='Sleuth') & (cast.year==1972) & (cast.n.notnull())].sort_values(by='n')

title  year                name     type  \
1644328  Sleuth  1972    Laurence Olivier    actor   
314691   Sleuth  1972       Michael Caine    actor   
360365   Sleuth  1972      Alec Cawthorne    actor   
1412647  Sleuth  1972  John (II) Matthews    actor   
2619278  Sleuth  1972  Eve (III) Channing  actress   
1396572  Sleuth  1972        Teddy Martin    actor   

                          character    n  
1644328                 Andrew Wyke  1.0  
314691                  Milo Tindle  2.0  
360365            Inspector Doppler  3.0  
1412647  Detective Sergeant Tarrant  4.0  
2619278             Marguerite Wyke  5.0  
1396572      Police Constable Higgs  6.0

### Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

In [21]:
cast[(cast.title=='Sleuth') & (cast.year==2007) & (cast.n.notnull())].sort_values(by='n')

title  year           name   type    character    n
314692   Sleuth  2007  Michael Caine  actor       Andrew  1.0
1246855  Sleuth  2007       Jude Law  actor         Milo  2.0
1741124  Sleuth  2007  Harold Pinter  actor  Man on T.V.  3.0

### How many roles were credited in the silent 1921 version of Hamlet?

In [22]:
print("Roles were creadited in 1921 of Hamlet movie-",len(cast[(cast.title=='Hamlet') & (cast.year==1921)]))

Roles were creadited in 1921 of Hamlet movie- 9


### How many roles were credited in Branagh’s 1996 Hamlet?

In [23]:
print("Roles were creadited in 1996 of Hamlet movie-",len(cast[(cast.title=='Hamlet') & (cast.year==1996)]))

Roles were creadited in 1996 of Hamlet movie- 54


### How many "Hamlet" roles have been listed in all film credits through history?

In [24]:
print("No. Of roles as Hamlet- ",len(cast[cast.character=="Hamlet"]))

No. Of roles as Hamlet-  91


### How many people have played an "Ophelia"?

In [25]:
print("No.of People Played as Ophelia - ",len(cast[cast.character=='Ophelia']))

No.of People Played as Ophelia -  111


### How many people have played a role called "The Dude"?

In [26]:
print("No.of People Played as THe Dude - ",len(cast[cast.character=='The Dude']))

No.of People Played as THe Dude -  18


### How many people have played a role called "The Stranger"?

In [27]:
print("No.of People Played as 'The Stranger' - ",len(cast[cast.character=='The Stranger']))

No.of People Played as 'The Stranger' -  208


### How many roles has Sidney Poitier played throughout his career?

In [28]:
print("No.Of roles played by Sidney Poitier - ",len(cast[cast.name=='Sidney Poitier']))

No.Of roles played by Sidney Poitier -  43


### How many roles has Judi Dench played?

In [29]:
print("No.of Roles Played by Judi Dench - ",len(cast[cast.name=='Judi Dench']))

No.of Roles Played by Judi Dench -  55


### List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

In [30]:
cast[(cast.name=='Cary Grant') & (cast.year//10==194) & (cast.n==2)].sort_values(by='year')

title  year        name   type    character    n
840371  My Favorite Wife  1940  Cary Grant  actor         Nick  2.0
840381    Penny Serenade  1941  Cary Grant  actor  Roger Adams  2.0

### List the leading roles that Cary Grant played in the 1940s in order by year.

In [31]:
cast[(cast.name=='Cary Grant') & (cast.year//10==194)].sort_values(by='year')

title  year        name   type  \
840399                The Philadelphia Story  1940  Cary Grant  actor   
840397               The Howards of Virginia  1940  Cary Grant  actor   
840353                       His Girl Friday  1940  Cary Grant  actor   
840371                      My Favorite Wife  1940  Cary Grant  actor   
840386                             Suspicion  1941  Cary Grant  actor   
840381                        Penny Serenade  1941  Cary Grant  actor   
840401                  The Talk of the Town  1942  Cary Grant  actor   
840377                 Once Upon a Honeymoon  1942  Cary Grant  actor   
840344                     Destination Tokyo  1943  Cary Grant  actor   
840369                             Mr. Lucky  1943  Cary Grant  actor   
840370                             Mr. Lucky  1943  Cary Grant  actor   
840336                  Arsenic and Old Lace  1944  Cary Grant  actor   
840378                      Once Upon a Time  1944  Cary Grant  actor   
840373             None But the Lonely Heart  1944  Cary Grant  actor   
840351               George White's Scandals  1945  Cary Grant  actor   
840372                         Night and Day  1946  Cary Grant  actor   
840375                             Notorious  1946  Cary Grant  actor   
840415                  Without Reservations  1946  Cary Grant  actor   
840393      The Bachelor and the Bobby-Soxer  1947  Cary Grant  actor   
840394                     The Bishop's Wife  1947  Cary Grant  actor   
840368  Mr. Blandings Builds His Dream House  1948  Cary Grant  actor   
840348          Every Girl Should Be Married  1948  Cary Grant  actor   
840357                I Was a Male War Bride  1949  Cary Grant  actor   

                    character    n  
840399     C. K. Dexter Haven  1.0  
840397            Matt Howard  1.0  
840353           Walter Burns  1.0  
840371                   Nick  2.0  
840386                Johnnie  1.0  
840381            Roger Adams  2.0  
840401           Leopold Dilg  1.0  
840377  Patrick 'Pat' O'Toole  1.0  
840344          Capt. Cassidy  1.0  
840369              Joe Adams  1.0  
840370        Joe Bascopolous  1.0  
840336      Mortimer Brewster  1.0  
840378            Jerry Flynn  1.0  
840373             Ernie Mott  1.0  
840351             Cary Grant  NaN  
840372            Cole Porter  1.0  
840375                 Devlin  1.0  
840415             Cary Grant  NaN  
840393            Dick Nugent  1.0  
840394                 Dudley  1.0  
840368          Jim Blandings  1.0  
840348      Dr. Madison Brown  1.0  
840357  Captain Henri Rochard  1.0

### How many roles were available for actors in the 1950s?

In [32]:
print("No.Of roles for Actors- ",len(cast[(cast.year//10==195) & (cast.type=='actor')]))

No.Of roles for Actors-  154727


### How many roles were available for actresses in the 1950s?

In [33]:
print("No.Of Roles for Actresses - ",len(cast[(cast.year//10==195) & (cast.type=='actress')]))

No.Of Roles for Actresses -  56259


### How many leading roles (n=1) were available from the beginning of film history through 1980?

In [34]:
print("No.of Leading roles upto 1980 - ",len(cast[(cast.n==1) & (cast.year<=1980)]))

No.of Leading roles upto 1980 -  63924


### How many non-leading roles were available through from the beginning of film history through 1980?

In [35]:
print("No.of Non-Leading roles upto 1980 - ",len(cast[(cast.n!=1) & (cast.year<=1980)]))

No.of Non-Leading roles upto 1980 -  1094780


### How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?

In [36]:
print("No.of minor roles upto 1980 and they did not rank- ",len(cast[(cast.n.isnull()) & (cast.year<=1980)]))

No.of minor roles upto 1980 and they did not rank-  438086
